### PR TITLE
V22F-518 File path database storage

### PR DESCRIPTION
**Pull request by noutheesink**
The path given to the database storage was not the same path as the buffer file. Fixed now

### CODE REVIEW

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]
Please merge with main to remove commits from the other PR

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl [hunk 1]
```diff
@@ -55,17 +55,18 @@ void DecisionMaker<SocketBlackBox, SDAConfig>::Initialize(tCarElt* p_initialCar,
...
-                                   blackBoxPath.filename().string(),
-                                   blackBoxPath.stem().string(),
-                                   std::chrono::system_clock::to_time_t(std::experimental::filesystem::last_write_time(blackBoxPath)),
-                                   p_track->filename,
-                                   p_track->name,
-                                   p_track->version,
-                                   Config.GetInterventionType());
+    m_bufferFilePath = m_fileBufferStorage.Initialize(
+        Config.GetDataCollectionSetting(),
+        BUFFER_FILE_PATH,
```
Comment (Sindarin27) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl:60]: The value of BUFFER_FILE_PATH should also be updated now that the temporary directory is properly used

**Review by Sindarin27 — CHANGES_REQUESTED** [review]

**Review by Sindarin27 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp [hunk 2]
```diff
@@ -121,7 +127,7 @@ void FileDataStorage::Save(tCarElt* p_car, tSituation* p_situation, unsigned lon
 
 /// @brief Save all decisions that were taken this tick
 /// @param p_decisions Tuple of decisions taken this tick
-void FileDataStorage::SaveDecisions(DecisionTuple& p_decisions)
+void FileDataStorage::SaveDecisions(DecisionTuple &p_decisions)
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp:130]: should be on left side

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp [hunk 3]
```diff
@@ -92,7 +98,7 @@ void FileDataStorage::Shutdown()
 /// @param p_car Current car status in Speed Dreams
 /// @param p_situation Current situation in Speed Dreams
 /// @param p_timestamp Current tick
-void FileDataStorage::Save(tCarElt* p_car, tSituation* p_situation, unsigned long p_timestamp)
+void FileDataStorage::Save(tCarElt *p_car, tSituation *p_situation, unsigned long p_timestamp)
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp:101]: should be on left side

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp [hunk 4]
```diff
@@ -34,23 +36,25 @@ inline void WriteTime(std::ostream& p_stream, time_t p_date)
...
-                                 const std::string& p_blackboxName,
-                                 const std::time_t& p_blackboxTime,
-                                 const std::string& p_environmentFilename,
-                                 const std::string& p_environmentName,
-                                 int p_environmentVersion,
-                                 InterventionType p_interventionType)
+/// @return returns the path of the buffer file
+std::experimental::filesystem::path FileDataStorage::Initialize(
+    tDataToStore p_saveSettings,
+    const std::string &p_fileName,
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp:42]: should be on left side

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]

**Review by KungFuDonkey — APPROVED** [review]

**Pull request merged**